### ISSUE TITLE
suggestions

### ISSUE THREAD

**Issue by andreasvc** [diasks2/pragmatic_segmenter#3]
Some other tools to consider for comparison:
- Elephant: http://www.aclweb.org/anthology/D13-1146
- English, regex-based: http://moin.delph-in.net/WeSearch/DocumentParsing
- Alpino is listed among the tools not yet tested. It is designed for Dutch and in my evaluation it came out as the best tokenizers/segmenter.

**Comment by diasks2** [comment]
Awesome, thank you! I will look into those. If you have any good example edge case tests to add for Dutch please let me know.

**Comment by andreasvc** [comment]
Here are a few:

Hij schoot op de JP8-brandstof toen de Surface-to-Air (sam)-missiles op
hem af kwamen. 81 procent van de schoten was raak. ...en toen barste de
hel los.

- JP8: hyphenated abbreviation with number
- (sam): parenthesized abbr.
- sentence starting with number
- sentence starting with ellipsis, after period
- quoted direct speech also caused issues in some tokenizers.

I forgot to mention ucto, a multilingual tokenizer:
http://ilk.uvt.nl/ucto/

**Comment by diasks2** [comment]
Thank you! I will work on adding those. Just to double check, I am assuming the text in your comment segments into 3 segments in the following way (please correct me if I am wrong):

```ruby
["Hij schoot op de JP8-brandstof toen de Surface-to-Air (sam)-missiles op
hem af kwamen.", "81 procent van de schoten was raak.", "...en toen barste de
hel los."]
```

**Comment by andreasvc** [comment]
Indeed.
Furthermore, in terms of tokenization, the hyphenated strings should
come out as a single token.

**Comment by diasks2** [comment]
Thanks. Right now this gem only handles sentence segmentation...but I hope to explore tokenization more in a future project.

**Comment by diasks2** [comment]
Thanks again for the great suggestions. :+1:  I updated the gem (https://github.com/diasks2/pragmatic_segmenter/commit/bcdd593d7f7b1e671342d5bf3f0663847cda62b6) with Dutch abbreviations and 2 specs based on what you mentioned:

* Sentence starting with a number
* Sentence starting with an ellipsis

The others you mentioned are more related to tokenization which is outside the scope of this gem, but I'll definitely keep them in mind for the tokenization gem I am working on. I also added the tools and papers you mentioned to the README. I haven't had time to test those tools yet, but if you have results from them I'd be happy to accept a pull request.
